d2networking/d2netpacket/packet_save_player.go

```diff
@@ -0,0 +1,44 @@
+package d2netpacket
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/OpenDiablo2/OpenDiablo2/d2core/d2map/d2mapentity"
+	"github.com/OpenDiablo2/OpenDiablo2/d2networking/d2netpacket/d2netpackettype"
+)
+
+// SavePlayerPacket has the actual selected left and right skill
+// the Server has to check if these skills are actually allowed for the Player
+type SavePlayerPacket struct {
+	LeftSkill  int `json:"leftSkill"`
+	RightSkill int `json:"rightSkill"`
+}
```

Comment:
We don't save any other data?

Reply:
The server already has got all of the base hero data. So we only need to send all of the things he can‘t possible know and need to write them back in its herodata Struct. There is much more we need to save, for example equipment data, but theres an extra issue for that #867 
So the structure will surely grow bigger.

We also need to think about when we send the Server our changed data and when it should save it to a file (or only update it‘s herodata without saving it). Surely on connection close (like in this code) that‘s a good idea to do. But maybe we need to send those packets more often, like when the equipment changes. The server not really does need to know which left and right skill is active on my client (it only needs to know if I have a skill when casting) but maybe it need to get notified everytime I change my equipment... I‘m not quite sure about this.
That sad, I think this code is a step into the right direction.
Thanks for your thoughts about this.